synfig-studio/src/gui/widgets/widget_timetrack.cpp

```diff
@@ -198,6 +198,40 @@ bool Widget_Timetrack::move_selected(synfig::Time delta_time)
 	return ok;
 }
 
+ void Widget_Timetrack::interpolate_selected(synfig::Interpolation type)
+{
+	synfig::Waypoint::Model model;
+	model.set_before(type);
+	model.set_after(type);
+	std::vector<WaypointItem*> selection = waypoint_sd.get_selected_items();
+
+	synfigapp::Action::PassiveGrouper group(get_canvas_interface()->get_instance().get(),_("Change Waypoint Group"));
```

Comment:
check if `selection` is empty before call the PassiveGrouper.
Otherwise the Undo history will have an empty action added.
Or does the PassiveGrouper self deletes if empty?

Reply:
I am not sure if it "self deletes" but clicking the buttons while nothing is selected doesnt make any new action and undo seems to be working as expected so I'm assuming it is taken care of ?

But similar to this I think I should check if the new interpolation is different or not to avoid making a new action and going through the methods if there is no change.

Reply:
> But similar to this I think I should check if the new interpolation is different or not to avoid making a new action and going through the methods if there is no change.

Indeed :)